core/tests/vm-benchmark/harness/src/lib.rs

```diff
@@ -96,33 +181,62 @@ impl BenchmarkingVm {
                 default_validation_computational_gas_limit: BATCH_COMPUTATIONAL_GAS_LIMIT,
                 chain_id: L2ChainId::from(270),
             },
-            &*STORAGE,
+            &STORAGE,
         ))
     }
+}
 
+impl<VM: BenchmarkingVmFactory> BenchmarkingVm<VM> {
     pub fn run_transaction(&mut self, tx: &Transaction) -> VmExecutionResultAndLogs {
         self.0.push_transaction(tx.clone());
         self.0.execute(VmExecutionMode::OneTx)
     }
 
+    pub fn run_transaction_full(&mut self, tx: &Transaction) -> VmExecutionResultAndLogs {
+        self.0.pop_snapshot_no_rollback();
+        self.0.make_snapshot();
+        let (compression_result, tx_result) = self.0.inspect_transaction_with_bytecode_compression(
+            Default::default(),
+            tx.clone(),
+            true,
+        );
+        compression_result.expect("compressing bytecodes failed");
+
+        if matches!(tx_result.result, ExecutionResult::Halt { .. }) {
+            self.0.rollback_to_the_latest_snapshot();
+        }
+        tx_result
+    }
+
     pub fn instruction_count(&mut self, tx: &Transaction) -> usize {
         self.0.push_transaction(tx.clone());
-
         let count = Rc::new(RefCell::new(0));
+        self.0.inspect(Default::default(), VmExecutionMode::OneTx); // FIXME
```

Comment:
Using yesterday's tracer improvements, this can be written as
```
struct InstructionCounter(usize);
impl<T> Tracer<T> for InstructionCounter {
    fn before_instruction(&mut self, ...) { self.0 += 1}
}
```

Reply:
This is "JFYI" at this point, right? Sure, it'll be possible to fix instruction counting once tracers are merged.

Reply:
Yes.